packages/transaction-controller/src/utils.ts

```diff
@@ -261,3 +264,39 @@ export function validateMinimumIncrease(proposed: string, min: string) {
   const errorMsg = `The proposed value: ${proposedDecimal} should meet or exceed the minimum value: ${minDecimal}`;
   throw new Error(errorMsg);
 }
+
+/**
+ * Helper Method to format transactions for the nonce tracker.
+ *
+ * @param fromAddress - Address of the account from which the transactions to filter from are sent.
+ * @param transactionStatus - Status of the transactions for which to filter.
+ * @param transactions - Array of transactionMeta objects that have been prefiltered.
+ * @returns Array of transactions formatted for the nonce tracker.
+ */
+export function getAndFormatTransactionsForNonceTracker(
```

Comment:
I opted to make this a utility function rather than a private method on the `TransactionsController` just so testing it made more sense... happy to move it to the controller if we think that makes more sense.